### ISSUE TITLE
Precision Recall metric for built in cross validation

### ISSUE THREAD

**Issue by mf-guavus** [Microsoft/LightGBM#2100]
## Environment info

Operating System: MacOS

CPU/GPU model: CPU

C++/Python/R version: Python

LightGBM version or commit hash: version 2.2.1

## Question

`r = lgb.cv(params, train_set, num_boost_round = 10000, nfold = 10, metrics = 'auc', 
           early_stopping_rounds = 100, verbose_eval = False, seed = 50)`

I have been trying to use Precision Recall AUC instead of ROC AUC as I am dealing with a high imbalanced data. I know that `metrics = 'auc'` means in fact the ROC AUC, Is it possible to add the PR AUC as well? What's the syntax to do so?

Thanks

**Comment by StrikerRUS** [comment]
Hi @mf-guavus !

You should use `feval` param for self-defined metrics.

Below you can find a toy example with a metric from scikit-learn:
```
import lightgbm as lgb

from sklearn.datasets import load_digits
from sklearn.metrics import average_precision_score


# proxy function for matching interfaces
def average_precision_score_(preds, train_data):
    y_true = train_data.get_label()
    return 'PR-AUC', average_precision_score(y_true, preds), True


train_set = lgb.Dataset(*load_digits(2, True))
params = {}
r = lgb.cv(params, train_set, num_boost_round = 10000, nfold = 10, metrics = 'auc', early_stopping_rounds = 100, verbose_eval = False, seed = 50, feval=average_precision_score_)
r
```


```
{'auc-mean': [0.9860456560791958,
  0.9818809005083515,
  0.9876724763979666,
  0.9846768336964417,
  0.9881445170660857,
  1.0],
 'auc-stdv': [0.024330364997497786,
  0.031448378092863,
  0.024154536760443792,
  0.0313611998353492,
  0.02521080567682969,
  0.0],
 'PR-AUC-mean': [0.9862656641604011,
  0.9863041863919056,
  0.9903954330270119,
  0.9900685425685426,
  0.9924007936507936,
  1.0],
 'PR-AUC-stdv': [0.02084886335609824,
  0.021447778293259976,
  0.0189306814894327,
  0.01986653284773583,
  0.01537474239789683,
  1.7199501139797033e-16]}
```

https://scikit-learn.org/stable/modules/generated/sklearn.metrics.average_precision_score.html
https://stats.stackexchange.com/questions/157012/area-under-precision-recall-curve-auc-of-pr-curve-and-average-precision-ap
